keps/sig-node/20190129-hugepages.md

```diff
@@ -231,22 +227,53 @@ mount -t hugetlbfs \
 ```
 
 The proposal recommends extending the existing `EmptyDirVolumeSource` to satisfy
-this use case.  A new `medium=HugePages` option would be supported.  To write
-into this volume, the pod must make a request for huge pages. The `pagesize`
-argument is inferred from the `hugepages-<hugepagesize>` from the resource
-request.  If in the future, multiple huge page sizes are supported in a single
-pod spec, we may modify the `EmptyDirVolumeSource` to provide an optional page
-size.  The existing `sizeLimit` option for `emptyDir` would restrict usage to
-the minimum value specified between `sizeLimit` and the sum of huge page limits
-of all containers in a pod. This keeps the behavior consistent with memory
-backed `emptyDir` volumes whose usage is ultimately constrained by the pod
-cgroup sandbox memory settings.  The `min_size` option is omitted as its not
-necessary.  The `nr_inodes` mount option is omitted at this time in the same
+this use case.  A new `medium=HugePages[-<hugepagesize>]` options would be
+supported.  To write into this volume, the pod must make a request for huge
+pages.  The `pagesize` argument is inferred from the `hugepages-<hugepagesize>`
+from the resource request.  The existing `sizeLimit` option for `emptyDir` would
+restrict usage to the minimum value specified between `sizeLimit` and the sum of
+huge page limits of all containers in a pod. This keeps the behavior consistent
+with memory backed `emptyDir` volumes whose usage is ultimately constrained by
+the pod cgroup sandbox memory settings.  The `min_size` option is omitted as its
+not necessary.  The `nr_inodes` mount option is omitted at this time in the same
 manner it is omitted with `medium=Memory` when using `tmpfs`.
 
-The following is a sample pod that is limited to 1Gi huge pages of size 2Mi. It
-can consume those pages using `shmget()` or via `mmap()` with the specified
-volume.
+The following is a sample pod that is limited to 1Gi huge pages of size 2Mi and
+2Gi huge pages of size 1Gi. It can consume those pages using `shmget()` or via
+`mmap()` with the specified volume.
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example
+spec:
+  containers:
+...
+    volumeMounts:
+    - mountPath: /hugepages-2Mi
+      name: hugepage-2Mi
+    - mountPath: /hugepages-1Gi
+      name: hugepage-1Gi
+    resources:
+      requests:
+        hugepages-2Mi: 1Gi
+        hugepages-1Gi: 2Gi
+      limits:
+        hugepages-2Mi: 1Gi
+        hugepages-1Gi: 2Gi
+  volumes:
+  - name: hugepage-2Mi
+    emptyDir:
+      medium: HugePages-2Mi
```

Comment:
I think adding the page size to the medium makes sense, and regret missing that prior.  For backwards compatibility, a pod that uses more than one page size needs to fail validation if a volume emptyDir medium=HugePages is specified.

Reply:
@derekwaynecarr thank you for the review!

Sorry for being unclear on this. There is an example of the invalid pod that was supposed to explain this particular case. I should've been more explicit I guess. I'll add more explanations to the PR.